MediaBrowser.MediaEncoding/Encoder/EncoderValidator.cs

```diff
@@ -153,9 +113,43 @@ private IEnumerable<string> GetEncoders(string encoderAppPath)
                 "ac3"
             };
 
-            var found = required.Where(x => output.IndexOf(x, StringComparison.OrdinalIgnoreCase) != -1);
+        private enum Codec
+        {
+            Encoder,
+            Decoder
+        }
+
+        private IEnumerable<string> GetCodecs(string encoderAppPath, Codec codec)
+        {
+            string codecstr = codec == Codec.Encoder ? "encoders" : "decoders";
+            string output = null;
+            try
+            {
+                output = GetProcessOutput(encoderAppPath, "-" + codecstr);
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, "Error detecting available {Codec}", codecstr);
+            }
+
+            if (string.IsNullOrWhiteSpace(output))
+            {
+                return Enumerable.Empty<string>();
+            }
+
+            var required = codec == Codec.Encoder ? requiredEncoders : requiredDecoders;
+
+            Regex regex = new Regex(@"\s\S{6}\s(?<codec>(\w|-)+)\s{3}");
```

Comment:
I'm not versed well enough in regexps, so a comment describing this one would be good.